contracts/core/libs/sismo-gated/SismoGated.sol

```diff
@@ -0,0 +1,156 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.14;
+import 'hardhat/console.sol';
+
+import {Badges} from '../../Badges.sol';
+import {Attester} from '../../Attester.sol';
+import {HydraS1AccountboundAttester} from '../../../attesters/hydra-s1/HydraS1AccountboundAttester.sol';
+import {HydraS1Base} from '../../../attesters/hydra-s1/base/HydraS1Base.sol';
+
+import {Request, Claim, Attestation} from '../Structs.sol';
+
+contract SismoGated {
+  // polygon contract address
+  address public constant BADGES_POLYGON_ADDRESS = 0xF12494e3545D49616D9dFb78E5907E9078618a34;
+  address public constant HYDRA_S1_ACCOUNTBOUND_POLYGON_ADDRESS =
+    0x10b27d9efa4A1B65412188b6f4F29e64Cf5e0146;
+
+  // goerli contract address
+  address public constant BADGES_GOERLI_ADDRESS = 0xE06B14D5835925e1642d7216F4563a1273509F10;
+  address public constant HYDRA_S1_ACCOUNTBOUND_GOERLI_ADDRESS =
+    0x89d80C9E65fd1aC8970B78A4F17E2e772030C1cB;
+
+  // mumbai contract address
+  address public constant BADGES_MUMBAI_ADDRESS = 0x5722fEa81027533721BA161964622271560da1aC;
+  address public constant HYDRA_S1_ACCOUNTBOUND_MUMBAI_ADDRESS =
+    0x069e6B99f4DA543156f66274FC6673442803C587;
+
+  // local contract address
+  address public constant BADGES_LOCAL_ADDRESS = 0xeF5b2Be9a6075a61bCA4384abc375485d5e196c3;
+  address public constant HYDRA_S1_ACCOUNTBOUND_LOCAL_ADDRESS =
+    0xf93A0C43A3466488D416628bf149495285e9f274;
+
+  uint8 public constant MAX_NUMBER_OF_BADGES = 10;
+
+  /**************************************
+   * Storage slot
+   * 20 slots
+   * -> 2 slots used
+   * -> 1 used by badges, 1 used by hydraS1AccountboundAttester
+   *
+   * -> 18 free slots
+   **************************************/
+
+  Badges public badges;
```

Comment:
Use immutable here

Reply:
In fact it's not possible, `Cannot write to immutable here: Immutable variables cannot be initialized inside an if statement`